src/Sparrow/Platform/Posix/SysUtils.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Text;
+using Sparrow.Logging;
+using Voron.Platform.Posix;
+
+namespace Sparrow.Platform.Posix
+{
+    public static class SysUtils
+    {
+        private static Logger _logger = LoggingSource.Instance.GetLogger("SysUtils", "Raven/Server");
+
+        public static unsafe ulong ReadULongFromFile(string filename)
+        {
+            var fd = Syscall.open(filename, OpenFlags.O_RDONLY, FilePermissions.S_IRUSR);
+            if (fd < 0)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Cannot open '{filename}'. Will not respect container's memory limit if running under one.");
+                return ulong.MaxValue;
+            }
+
+            ulong cgroup;
+            UIntPtr readSize = (UIntPtr)32;
+            IntPtr pBuf = Marshal.AllocHGlobal((int)readSize);
+            try
+            {
+                Memory.Set((byte*)pBuf, 0, 32);
+                var cgroupRead = Syscall.read(fd, pBuf.ToPointer(), (ulong)readSize);
+                if (cgroupRead > 30 || cgroupRead == 0) // check we are not garbadged
+                {
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"Got invalid number of characters ({cgroupRead}) from filename. Will not respect container's memory limit if running under one.");
+                    Syscall.close(fd);
+                    return ulong.MaxValue;
+                }
+
+                Syscall.close(fd);
+
+                string str = null;
+                try
+                {
+                    str = Encoding.ASCII.GetString((byte*)pBuf.ToPointer(), (int)cgroupRead);
+                    cgroup = Convert.ToUInt64(str);
+                }
+                catch (Exception ex)
+                {
+
+                    if (_logger.IsInfoEnabled)
+                        _logger.Info($"couldn't convert string '{str}' to long. Will not respect container's memory limit if running under one.", ex);
+                    cgroup = ulong.MaxValue;
+                }
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(pBuf);
+            }
+            return cgroup;
+        }
+
+        public static unsafe List<int> ReadAndParseRangesFromFile(string filename)
+        {
+            var fd = Syscall.open(filename, OpenFlags.O_RDONLY, FilePermissions.S_IRUSR);
+            if (fd < 0)
+            {
+                if (_logger.IsInfoEnabled)
+                    _logger.Info($"Cannot open '{filename}'. Will not respect container's number of cores if running under one.");
```

Comment:
What is this error here if you are passing a `filename` ? This assume that you know what you are calling, and either should be hard coded or generic.